src/main.rs

```diff
@@ -248,6 +262,18 @@ async fn main() {
             app.add_to_log(format!("Failed to load channels: {}", e));
         }
     }
+
+    // Load initial unread counts and update UI
+    match get_unread_counts_by_channel(&PEER_ID.to_string()).await {
```

Comment:
This code is duplicated in many places in main.rs.  It would be a good idea to create a helper function that would remove the duplicated code.